examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt

```diff
@@ -355,6 +380,7 @@ class SimpleMapboxNavigationKt : AppCompatActivity(), OnMapReadyCallback {
         if (mapboxNavigation.getRoutes().isEmpty() && mapboxNavigation.getTripSessionState() == TripSessionState.STARTED) {
             // use this to kill the service and hide the notification when going into the background in the Free Drive state,
             // but also ensure to restart Free Drive when coming back from background by using the channel
+            mapboxNavigation.unregisterVoiceInstructionsObserver(this)
```

Comment:
what if put register/unregister to start/stop _trip session_ `observer`(to observe it register observer `mapboxNavigation.registerTripSessionStateObserver()`). wdyt?

Reply:
in that case, do we kind of lose the flexibility? some customers may only interest in voice instruction and some others may only care banner view update etc.?

Reply:
IMO it looks not so obvious when register/unregister observer could register/unregister from a few places. But it's samples and I'm ok to keep it as is. up 2 you 👍